include/xgboost/data.h

```diff
@@ -47,6 +47,12 @@ class MetaInfo {
   uint64_t num_nonzero_{0};
   /*! \brief label of each instance */
   HostDeviceVector<bst_float> labels_;
+  /*! \brief lower bound label of each instance; used for survival analysis,
+   *         where labels are left-, right- or interval-censored. */
+  HostDeviceVector<bst_float> labels_lower_bound_;
```

Comment:
> There are huge implications to doing this.

Please be more specific. Didn't a recent PR already remove `qids_` field from MetaInfo?

> Is there anyway to represent this information in the labels vector?

We need both lower and upper bounds, so that we can support different kinds of censored labels. The idea is to specify a range in which a label falls in.

Reply:
@RAMitchell Basically, I would like to associate two extra 1d arrays with DMatrix objects. If modifying the MetaInfo class is not desired, can we at least add a generic interface to associate an arbitrary 1d array with DMatrix objects?

Reply:
One way of looking at this is asking, what would happen if we added arbitrary meta info for another 5 unique loss functions over the next few years? Perhaps a key value store in MetaInfo implemented as std::map<std::string,HostDeviceVector<float>>? What about passing the labels as a 2x length vector? Not sure I necessarily like this approach but it's similar to how we deal with gradients for multiclass. The correct label length would have to be checked in the objective and metric.

Reply:
> what would happen if we added arbitrary meta info for another 5 unique loss functions over the next few years?

Okay, that’s a fair point. A key-value storage will certainly result in less clutter. It turns out that @avinashbarnwal is planning to add the binomial loss, which requires a 1d array to represent the number of trials.

Would you be satisfied with std::map<std::string, HostDeviceVector> ?